utils/saasreq/request.go

```diff
@@ -63,6 +66,10 @@ func MakeRequest(ctx context.Context, method string, endpoint, accessToken strin
 		h.Add("Authorization", fmt.Sprintf("Bearer %s", accessToken))
 	}
 
+	if RequestEnricher != nil {
+		RequestEnricher(method, endpoint, req)
+	}
+
```

Comment:
extension point for SaaS req headers enrichment, so that we can call telemetryD directly without API gateway